webapp/app/components/pages/login/LoginHOC.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as authActions from '../../../actions/authActions';
+
+const LoginHOC = (LoginChild) => {
+  class LoginParent extends React.Component {
+    static propTypes = {
+      actions: PropTypes.object.isRequired
+    };
+
+    constructor() {
+      super();
+
+      this.state = {
+        username: '',
+        password: '',
+        auth: {}
+      };
+    }
+
+    handleOnChange = (e) => {
+      const { name, value } = e.target;
+
+      this.setState({
```

Comment:
we should add at least a couple of basic checks, at least if `name` is not falsy, and if the value is different than the prior state value for that name key, only then updat ethe state for that key with setState, 
otherwise we endup updating unecesarily a key with the same exact value in the state,
and in the unlickely case the name is undefined or empty, we endup invoking setStat e with invalid keys or causing exceptions.

Perhaps the use of something along these lines would be more efficient
```
if (name && this.state[name] !== value) {
    this.setState({
        [name]: value
    });
}
```